examples/scene_subplots_video.py

```diff
@@ -0,0 +1,144 @@
+"""
+An example combining `synced_video.py` with subplots
+"""
+
+from wgpu.gui.auto import WgpuCanvas, run
+import pygfx as gfx
+import numpy as np
+
+canvas = WgpuCanvas()
+renderer = gfx.renderers.WgpuRenderer(canvas)
+
+dims = (512, 512)  # image dimensions
+# default cam position
+center_cam_pos = (256, 256, 0)
+
+# colormaps for each of the 4 images
+cmaps = [gfx.cm.inferno, gfx.cm.plasma, gfx.cm.magma, gfx.cm.viridis]
+
+# lists of everything necessary to make this plot
+scenes = list()
+cameras = list()
+images = list()
+controllers = list()
+cntl_defaults = list()
+viewports = list()
+
+for i in range(4):
+    # create scene for this subplot
+    scene = gfx.Scene()
+    scenes.append(scene)
+
+    # create Image WorldObject
+    img = gfx.Image(
+        gfx.Geometry(
+            grid=gfx.Texture(np.random.rand(*dims).astype(np.float32) * 255, dim=2)
+        ),
+        gfx.ImageBasicMaterial(clim=(0, 255), map=cmaps[i]),
+    )
+
+    # add image to list
+    images.append(img)
+    scene.add(img)
+
+    # create camera, set default position, add to list
+    camera = gfx.OrthographicCamera(*dims)
+    camera.position.set(*center_cam_pos)
+    cameras.append(camera)
+
+    # create viewport for this image
+    viewport = gfx.Viewport(renderer)
+    viewports.append(viewport)
+
+    # controller for pan & zoom
+    controller = gfx.PanZoomController(camera.position.clone())
+    controller.add_default_event_handlers(viewport, camera)
+    controllers.append(controller)
+
+    # get the initial controller params so the camera can be reset later
+    cntl_default = dict()
+    cntl_default["distance"] = controller.distance
+    cntl_default["zoom_value"] = controller.zoom_value
+    cntl_default["target"] = controller.target.clone()
+    cntl_defaults.append(cntl_default)
+
+w_div = 2
+h_div = 2
+
+
+# 2x2 arrangement for the viewports
+def produce_rect(w, h):
+    return [
+        (0, 0, w / w_div, h / h_div),
+        (w / w_div, 0, w / w_div, h / h_div),
+        (0, h / h_div, w / w_div, h / h_div),
+        (w / w_div, h / h_div, w / w_div, h / h_div),
+    ]
+
+
+reset_cameras = False
+
+
+def animate():
+    for img in images:
+        # create new image data
+        img.geometry.grid.data[:] = np.random.rand(*dims).astype(np.float32) * 255
+        img.geometry.grid.update_range((0, 0, 0), img.geometry.grid.size)
+        # img.geometry.grid = gfx.Texture(np.random.rand(*dims).astype(np.float32) * 255, dim=2)
+
+    w, h = canvas.get_logical_size()
+
+    rects = produce_rect(w, h)
```

Comment:
I would recommend updating the viewport rects in a separate function using the canvas resize event, for an example of how to achieve this see:

https://github.com/pygfx/pygfx/blob/main/examples/scene_subplots2.py#L79-L86

Reply:
Thanks, is there anything particularly not-recommended about updating the viewports like this? In `fastplotlib` I'm using this method for updating the viewports for an arbitrary `i x j` grid:

```python
def _produce_rect(j, i, ncols, nrows, w, h):
    return [
        ((w / ncols) + ((i - 1) * (w / ncols))),
        ((h / nrows) + ((j - 1) * (h / nrows))),
        (w / ncols),
        (h / nrows)
    ]
```

https://github.com/kushalkolar/fastplotlib/blob/796f291086703cb752836d69d1ef2b52cfdf337d/fastplotlib/layouts.py#L9

Reply:
Just performance optimization. It's code that technically doesn't need to run on every frame. But you're free to choose to do it anyway.

Reply:
ah makes sense, didn't realize that `renderer.add_event_handler` is like a callback signal (sorry I'm coming from years of working in Qt). I think I can generalize it with the event handler for an `i x j` grid.


anyways I'll update this PR soon with the even handler!